stats/influxdb/collector.go

```diff
@@ -133,8 +149,26 @@ func (c *Collector) commit() {
 }
 
 func (c *Collector) extractTagsToValues(tags map[string]string, values map[string]interface{}) map[string]interface{} {
+tags:
 	for _, tag := range c.Config.TagsAsFields {
 		if val, ok := tags[tag]; ok {
+			if kind, convNeeded := c.fieldKinds[tag]; convNeeded {
+				var v interface{}
+				var err error
+				switch kind {
+				case Bool:
+					v, err = strconv.ParseBool(val)
+				case Float:
+					v, err = strconv.ParseFloat(val, 64)
+				case Int:
+					v, err = strconv.ParseInt(val, 10, 64)
+				}
+				if err == nil {
+					values[tag] = v
+					delete(tags, tag)
+					continue tags
+				}
+			}
```

Comment:
Can you add some tests that cover these changes? Ideally we should have more functional InfluxDB tests, but this method looks like it can be unit tested.

Reply:
@imiric Thank you very much for your comments.

> Though in general I would've preferred if this didn't require as many separate environment variables, and maybe enhanced the existing K6_INFLUXDB_TAGS_AS_FIELDS to accept type information. Something like K6_INFLUXDB_TAGS_AS_FIELDS=vu:bool,iter:bool,mytag:int. Not a blocker, just a preference. :)

I have to admit that it's a clever design especially for those field names need to be defined in K6_INFLUXDB_TAGS_AS_FIELDS (otherwise they are stored as tags).  I'll update the code using your design.

Reply:
Added a new test case to cover that part of code.